### PR TITLE
Fix the comparing boundary points of `concept-range-bp-set`

### LINKED ISSUE #924: concept-range-bp-set should compare root nodes before bp relative position
In https://dom.spec.whatwg.org/#concept-range-bp-set:

> If bp is after the range’s end, or if range’s root is not equal to node’s root, set range’s end to bp.

(and similar for "set the end")

This implies that implementations should check if "bp is after the range’s end", however the first step of comparing boundary points is to "Assert: nodeA and nodeB have the same root.".

I would suggest this be re-ordered to:

> If range’s root is not equal to node’s root, or if bp is after the range’s end, set range’s end to bp.

(and similar for "set the end")

### PR DESCRIPTION
**Pull request by chansuke**
Fixes #924

***
<a href="https://whatpr.org/dom/925.html" title="Last updated on Nov 17, 2020, 11:36 PM UTC (44eecc3)">Preview</a> | <a href="https://whatpr.org/dom/925/52f7a52...44eecc3.html" title="Last updated on Nov 17, 2020, 11:36 PM UTC (44eecc3)">Diff</a>

### CODE REVIEW

**Review by annevk — APPROVED** [review]
Thanks, this looks good. @edg2s FYI.

**Comment by annevk** [comment]
@chansuke if you want you can add your name to the Acknowledgments section.

**Comment by edg2s** [comment]
+1

File: dom.bs [hunk 1]
```diff
@@ -7803,28 +7803,28 @@ steps:
...
-     <var>range</var>'s <a for=range>start</a>,
-     or if <var>range</var>'s
+     <li>If <var>range</var>'s
      <a for="live range">root</a> is not equal to
-     <var>node</var>'s <a for=tree>root</a>, set
-     <var>range</var>'s <a for=range>start</a>
+     <var>node</var>'s <a for=tree>root</a>,
+     or if <var>bp</var> is
+     <a for="boundary point">before</a> the
+     <var>range</var>'s <a for=range>end</a>, set
```
Comment (edg2s) [inline dom.bs:7826]: this should be "start"

File: dom.bs [hunk 2]
```diff
@@ -7803,28 +7803,28 @@ steps:
...
+     <var>range</var>'s <a for=range>end</a>
```
Comment (edg2s) [inline dom.bs:7827]: this should be "start"

**Review by edg2s — CHANGES_REQUESTED** [review]

**Comment by edg2s** [comment]
"start" was inadvertently changed to "end", presumably and copy-paste error, as can be seen in the HTML diff:
image

**Comment by chansuke** [comment]
@edg2s (CC: @annevk )I appreciate your review. I fixed and rebased it(since it's a small change).

**Review by annevk — APPROVED** [review]

**Comment by annevk** [comment]
Thanks all!

**Pull request merged**